libbeat/cmd/instance/metrics.go

```diff
@@ -106,36 +106,81 @@ func reportBeatCPU(_ monitoring.Mode, V monitoring.Visitor) {
 	V.OnRegistryStart()
 	defer V.OnRegistryFinished()
 
-	totalCPUUsage, err := getCPUUsage()
+	totalCPUUsage, totalCPUUserTicks, totalCPUSystemTicks, totalCPUTicks, err := getCPUUsage()
 	if err != nil {
 		logp.Err("Error retrieving CPU percentages: %v", err)
 		return
 	}
 
+	userTime, systemTime, err := process.GetOwnResourceUsageTime()
+	if err != nil {
+		logp.Err("Error retrieving CPU usage time: %v", err)
+		return
+	}
+
+	monitoring.ReportNamespace(V, "user", func() {
+		monitoring.ReportInt(V, "ticks", int64(totalCPUUserTicks))
+		monitoring.ReportInt(V, "time", userTime)
+	})
+	monitoring.ReportNamespace(V, "system", func() {
+		monitoring.ReportInt(V, "ticks", int64(totalCPUSystemTicks))
+		monitoring.ReportInt(V, "time", systemTime)
+	})
 	monitoring.ReportNamespace(V, "total", func() {
 		monitoring.ReportFloat(V, "value", totalCPUUsage)
+		monitoring.ReportInt(V, "ticks", int64(totalCPUTicks))
+		monitoring.ReportInt(V, "time", userTime+systemTime)
 	})
-
 }
 
-func getCPUUsage() (float64, error) {
+func getCPUUsage() (float64, uint64, uint64, uint64, error) {
```

Comment:
return tuple is quite big. Some naming (e.g. introduce a new type) would make the result more parseable.